### PR TITLE
You can now ruin a perfectly good bag of chips by stepping on them

### PR DESCRIPTION
**Pull request by Ryll-Ryll**
## About The Pull Request
Stepping on a fresh untouched bag of chips will pop the bag and ruin all the delicious crunchiness inside. Why would you do this?

https://cdn.discordapp.com/attachments/326831214667235328/736428578438119475/Luna_Array_Seven_2020-07-24_23-41-47.mp4
need to find a better sound since pop.ogg is giving me trouble

## Why It's Good For The Game
adds immersion

## Changelog
:cl: Ryll/Shaps
add: Stepping on an unopened bag of chips will pop it and ruin all the chips inside
/:cl:

### CODE REVIEW

File: code/modules/food_and_drinks/food/snacks_vend.dm [hunk 1]
```diff
@@ -77,6 +77,20 @@
...
+	if(!isliving(AM) || bitecount) // can't pop opened chips
+		return
+
+	var/mob/living/popper = AM
+	if(popper.mob_size < MOB_SIZE_HUMAN)
+		return
+
+	playsound(get_turf(src), 'sound/effects/pop_expl.ogg', 50, FALSE)
+	popper.visible_message("<span class='danger'>[popper] steps on \the [src], popping the bag!</span>", "<span class='danger'>You step on \the [src], popping the bag!</span>", "<span class='danger'>You hear a gunshot!</span>", COMBAT_MESSAGE_RANGE)
+	generate_trash(src.loc)
```
Comment (ShizCalev) [inline code/modules/food_and_drinks/food/snacks_vend.dm:91]: just loc

File: code/modules/food_and_drinks/food/snacks_vend.dm [hunk 2]
```diff
@@ -77,6 +77,20 @@
...
+/obj/item/reagent_containers/food/snacks/chips/Crossed(atom/movable/AM, oldloc)
+	. = ..()
+	if(!isliving(AM) || bitecount) // can't pop opened chips
+		return
+
+	var/mob/living/popper = AM
+	if(popper.mob_size < MOB_SIZE_HUMAN)
+		return
+
+	playsound(get_turf(src), 'sound/effects/pop_expl.ogg', 50, FALSE)
```
Comment (ShizCalev) [inline code/modules/food_and_drinks/food/snacks_vend.dm:89]: src

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Krysonism** [comment]
Why doesn't this work on boritos corn chips as well? Immersion ruined.

boritos_comp

**Comment by Dorsisdwarf** [comment]
Needs to spray the crisps all across the floor and give you the ability to pick up and eat the stray crisps but anyone who sees you do it pukes and gets a negative moodlet whenever they see you for the rest of the round

**Pull request merged**